Xplat/src/main/java/vazkii/botania/common/block/flower/functional/PollidisiacBlockEntity.java

```diff
@@ -35,37 +43,75 @@ public PollidisiacBlockEntity(BlockPos pos, BlockState state) {
 	public void tickFlower() {
 		super.tickFlower();
 
-		if (!getLevel().isClientSide) {
-			var pickupBounds = new AABB(getBlockPos().offset(-RANGE, -RANGE, -RANGE), getBlockPos().offset(RANGE + 1, RANGE + 1, RANGE + 1));
-			List<ItemEntity> items = getLevel().getEntitiesOfClass(ItemEntity.class, pickupBounds);
-			var bounds = new AABB(getEffectivePos().offset(-RANGE, -RANGE, -RANGE), getEffectivePos().offset(RANGE + 1, RANGE + 1, RANGE + 1));
-			List<Animal> animals = getLevel().getEntitiesOfClass(Animal.class, bounds);
+		if (!getLevel().isClientSide && getMana() >= MANA_COST) {
+			var pickupBounds = new AABB(getBlockPos()).inflate(RANGE);
+			List<ItemEntity> items = getLevel().getEntitiesOfClass(ItemEntity.class, pickupBounds,
+					itemEntity -> DelayHelper.canInteractWith(this, itemEntity));
+			if (items.isEmpty()) {
+				return;
+			}
+			var bounds = new AABB(getEffectivePos()).inflate(RANGE);
+			List<Animal> animals = getLevel().getEntitiesOfClass(Animal.class, bounds, Predicate.not(Animal::isBaby));
+			Collections.shuffle(animals);
 
 			for (Animal animal : animals) {
-				if (getMana() < MANA_COST) {
-					break;
+				// Note: Empty item stacks are implicitly excluded in Animal::isFood and ItemStack::is(TagKey)
+				if (!animal.isInLove()) {
+					for (ItemEntity item : items) {
```

Comment:
I'm not quite sure this works as intended. Won't this consume multiple items on one animal? There's nothing that breaks out of the loop once a food item has been consumed from what i can see.